compiler/rustc_session/src/filesearch.rs

```diff
@@ -82,9 +82,7 @@ fn current_dll_path() -> Result<PathBuf, String> {
                 let fname_ptr = info.dli_fname.as_ptr();
                 #[cfg(not(target_os = "cygwin"))]
                 let fname_ptr = {
-                    if info.dli_fname.is_null() {
-                        return Err("dladdr returned null pointer".into());
-                    }
+                    assert!(!info.dli_fname.is_null(), "the docs do not allow dladdr to be null");
```

Comment:
maybe "do not allow *dli_fname* to be null" ?  as you pointed out, dladdr can return 0, and `dli_sname` can be null.

Reply:
*oops*